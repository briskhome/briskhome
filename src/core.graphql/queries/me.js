/** @flow
 * @briskhome
 * â””core.graphql <queries/me.js>
 */

import UserType from '../types/User';
import type { Context } from '../../utilities/coreTypes';

export default {
  type: UserType,
  resolve: async (src: Object, args: {}, ctx: Context) => {
    const { db, req: { user } } = ctx;
    const UserModel = db.model('core:user');
    if (!user) return null;
    return UserModel.fetchByUsername(user.username, { lean: true });
  },
};
