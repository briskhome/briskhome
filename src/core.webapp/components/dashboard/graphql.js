import {gql} from 'react-apollo';

export const usersQuery = gql`
  query users {
    users {
      id
      name
    }
  }
`;
