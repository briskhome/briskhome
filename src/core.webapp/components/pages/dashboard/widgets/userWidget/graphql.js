import { gql } from 'react-apollo';

export const users = gql`
  query users {
    users {
      username
      name
      type
      isActive
    }
  }
`;

export const createUser = gql`
  mutation createUser(
    $lastName: String!
    $firstName: String!
    $password: String!
    $type: UserTypeEnum!
  ) {
    createUser(
      input: {
        lastName: $lastName
        firstName: $firstName
        password: $password
        type: $type
      }
    ) {
      firstName
      lastName
      username
    }
  }
`;

export const removeUser = gql`
  mutation removeUser($username: String!) {
    removeUser(input: { username: $username }) {
      result
    }
  }
`;
