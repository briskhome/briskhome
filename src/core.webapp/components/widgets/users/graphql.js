import { gql } from 'react-apollo';

export const users = gql`
  query users {
    users {
      id
      name
      type
    }
  }
`;

// export const createUser = gql`
//   mutation createUser(lastName: String!, firstName: String!, type: UserTypeEnum!) {
//     lastName: $lastName, firstName: $firstName, type: $type {
//       id, name
//     }
//   }
// `;
