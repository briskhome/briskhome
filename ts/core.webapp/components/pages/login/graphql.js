/** @flow */
import { gql } from 'react-apollo';
export const login = gql`
  mutation login($username: String!, $password: String!) {
    login(input: { username: $username, password: $password }) {
      firstName
      lastName
      username
      type
    }
  }
`;