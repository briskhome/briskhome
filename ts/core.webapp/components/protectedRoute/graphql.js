import { gql } from 'react-apollo';
export const me = gql`
  query {
    me {
      firstName
      lastName
      username
      type
    }
  }
`;