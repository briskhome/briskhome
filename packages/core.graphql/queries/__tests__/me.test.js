import { graphql, GraphQLSchema, GraphQLObjectType } from 'graphql';
import me from '../me';

const data = {
  username: 'jdoe',
  name: 'John Doe',
  type: 'regular',
};

const query = `query {
  me {
    username
    name
    type
  }
}`;

describe('core.graphql -> queries -> me', () => {
  let schema;
  const mockDb = {
    model: jest.fn(() => ({
      fetchByUsername: jest.fn(() => data),
    })),
  };

  beforeEach(() => {
    schema = new GraphQLSchema({
      query: new GraphQLObjectType({
        name: 'RootQueryType',
        fields: {
          me,
        },
      }),
    });
  });

  it('query', async () => {
    expect(
      await graphql(schema, query, null, { db: mockDb, req: { user: data } }),
    ).toMatchSnapshot();
  });

  it('anonymous query', async () => {
    expect(
      await graphql(schema, query, null, { db: mockDb, req: { user: null } }),
    ).toMatchSnapshot();
  });
});
