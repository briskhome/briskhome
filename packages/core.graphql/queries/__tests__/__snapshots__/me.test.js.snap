// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.graphql -> queries -> me anonymous query 1`] = `
Object {
  "data": Object {
    "me": null,
  },
}
`;

exports[`core.graphql -> queries -> me query 1`] = `
Object {
  "data": Object {
    "me": Object {
      "name": "John Doe",
      "type": "regular",
      "username": "jdoe",
    },
  },
}
`;
